src/global.cpp

```diff
@@ -88,6 +93,61 @@ std::shared_future<void> Cleanup() { return impl::Init::Instance().cleanup(); }
 
 void SetSctpSettings(SctpSettings s) { impl::Init::Instance().setSctpSettings(std::move(s)); }
 
+std::map<int, UnhandledStunRequestHandler *> unboundStunCallbacks;
+
+#if !USE_NICE
+
+void InvokeUnhandledStunRequestCallback (const juice_mux_binding_request *info, void *user_ptr) {
+	PLOG_DEBUG << "Invoking unhandled STUN request callback";
+
+	UnhandledStunRequestHandler *handler = (struct UnhandledStunRequestHandler*)user_ptr;
+
+	if (handler->callback) {
+		handler->callback({
+			std::string(info->local_ufrag),
+			std::string(info->remote_ufrag),
+			std::string(info->address),
+			info->port
+		}, handler->userPtr);
+	} else {
+		PLOG_DEBUG << "No unhandled STUN request callback configured for port " << info->port;
+	}
+}
+
+#endif
+
+void OnUnhandledStunRequest ([[maybe_unused]] std::string host, [[maybe_unused]] int port, [[maybe_unused]] UnhandledStunRequestCallback callback, [[maybe_unused]] void *userPtr) {
+	#if USE_NICE
+		PLOG_WARNING << "BindStunListener is not supported with libnice, please use libjuice";
+	#else
+	if (callback == NULL) {
+		PLOG_DEBUG << "Removing unhandled STUN request listener";
+
+		free(unboundStunCallbacks.at(port));
```

Comment:
This causes a race condition: If the callback is called in parallel it will cause a use after free.

Also, if no callback is set for the port it will throw an `std::out_of_range` exception which is confusing for the user so you should use `find` instead.

Reply:
I've removed the map - `ListenIceUdpMux` now accepts a pointer to a callback and the caller of `ListenIceUdpMux` is responsible for it's memory management.